### PR TITLE
Enable TensorDataset to get any number of tensors

### PR DESCRIPTION
**Pull request by jaesuny**
Keeping compatibility, enable `TensorDataset` to get any number of tensors.

### CODE REVIEW

**Comment by ezyang** [comment]
This needs a test

File: torch/utils/data/dataset.py [hunk 1]
```diff
@@ -32,18 +32,27 @@ class TensorDataset(Dataset):
...
-        self.target_tensor = target_tensor
+    def __init__(self, data_tensor=None, target_tensor=None, *tensors):
+        def tensor_gen():
+            if data_tensor is not None:
+                yield data_tensor
+            if target_tensor is not None:
+                yield target_tensor
+            for tensor in tensors:
+                yield tensor
+            pass
```
Comment (apaszke) [inline torch/utils/data/dataset.py]: This `pass` is unnecessary

File: torch/utils/data/dataset.py [hunk 2]
```diff
@@ -32,18 +32,27 @@ class TensorDataset(Dataset):
...
         data_tensor (Tensor): contains sample data.
         target_tensor (Tensor): contains sample targets (labels).
+        *tensors (Tensor): tensors with the same first dimension.
     """
 
-    def __init__(self, data_tensor, target_tensor):
-        assert data_tensor.size(0) == target_tensor.size(0)
-        self.data_tensor = data_tensor
-        self.target_tensor = target_tensor
+    def __init__(self, data_tensor=None, target_tensor=None, *tensors):
```
Comment (apaszke) [inline torch/utils/data/dataset.py]: I don't think anyone has ever used those kwargs. Let's simplify the signature and take in `*tensors`.

Comment (Jaesuny) [reply torch/utils/data/dataset.py]: That would make it simpler.

**Review by apaszke — APPROVED** [review]

**Pull request merged**